### ISSUE TITLE
ArgoCD: Fluentd helm chart deployment error due to deprecated HorizontalPodAutoScaler api

### ISSUE THREAD

**Issue by ricsanfre** [ricsanfre/pi-cluster#92]
ArgoCD not able to get status of logging application with the following error:

![image](https://user-images.githubusercontent.com/84853324/212466671-e2d30f8a-e2e3-4c52-9005-7ec767a55992.png)

HPA is configured only when deploying fluentd. The manifest used by fluentd helm chart ([hpa.yml](https://github.com/fluent/helm-charts/blob/main/charts/fluentd/templates/hpa.yaml)) is using a deprecated version of autscaling API (autoscaling/v2) since v1.23+

```shell
kubectl describe HorizontalPodAutoscaler logging-fluentd -n logging
Warning: autoscaling/v2beta2 HorizontalPodAutoscaler is deprecated in v1.23+, unavailable in v1.26+; use autoscaling/v2 HorizontalPodAutoscaler
```
```shell
kubectl get HorizontalPodAutoscaler logging-fluentd -o yaml -n logging
```

Even when original manifest file is using a deprecated version (autoscaling/v2beta2), kubernetes is applying the highest API version available for autoscaling which is autoscaling/v2.

```yml
apiVersion: autoscaling/v2
kind: HorizontalPodAutoscaler
metadata:
  creationTimestamp: "2023-01-14T09:09:03Z"
  labels:
    app.kubernetes.io/instance: logging
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: fluentd
    app.kubernetes.io/version: v1.14.6
    argocd.argoproj.io/instance: logging
    helm.sh/chart: fluentd-0.3.9
  name: logging-fluentd
  namespace: logging
  resourceVersion: "14605"
  uid: 8b32c4f6-261b-4cfd-bced-f491eff04d75
spec:
  maxReplicas: 100
  metrics:
  - resource:
      name: cpu
      target:
        averageUtilization: 80
        type: Utilization
    type: Resource
  minReplicas: 1
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: logging-fluentd
```

It seems that ArgoCD is not able to calculate the differences between what is really applied to the cluster (using autoscaling/v2 api) and the original manifest file (using autoscaling/v2beta2)

**Comment by ricsanfre** [comment]
ArgoCD sync initial issue with Loki's statefulset resources has been solved in release 2.10, when 
Implement Server-Side Diff in Argo CD controller (https://github.com/argoproj/argo-cd/issues/11574)

**Comment by ricsanfre** [comment]
Keeping this issue for tracking fluentd helm chart issue

**Comment by ricsanfre** [comment]
Original fluend helm chart issue fixed with release 0.5.2
